### ISSUE TITLE
Test image+video+audio+audio/video T2AV training

### ISSUE THREAD

**Issue by AkaneTendo25** [LongCatFan1/longcat-tuner#61]
Need, for once, to do a full-modality ("freeform") LoRA fine-tuning to confirm that no subtle bugs emerge.

**Comment by randomkaos-boop** [comment]
I'll be watching this for T2V image training. I've not had much success so far with it. Samples when training look great. Awesome likeness, Inference on ComfyUI with base + distilled lora = 40% likeness or so. Even without the distilled lora, and settling for a bit worse quality it still stays around 40%. Tried multiple LR's, ranks and datasets so far, but just cant seem to get it to the right level. Interested to know if you fair any better and what settings you are using - hell ill even take captioning samples at this point!

**Comment by dorpxam** [comment]
My purely subjective observation, based on my initial "estimate" tests, is that the default audio-video training mode seems to overemphasize audio compared to video. Perhaps the default settings should be balanced while still allowing advanced users to customize their adjustments.

**Comment by randomkaos-boop** [comment]
I think I may be doing something wrong, can someone cast their eye over, spot anything glaringly obvious that I missed please.

If I train a lora - 2e-4 LR, 600 steps, 512 to 1024 resolutions in fp16 - the produced lora I then add to a comfyui workflow at 1 strength - its started to learn the subject. If i run that same lora at 1.8 to 2 strength, its actually very nearly spot on likeness.

Interestingly, on a previous run same settings, but using fp8 base and scaled, same result, little to no match by epoch 40 (920 steps) but is "learning" at strength 1. Strength 2... very nearly spot on - pretty much same face as above. (same dataset ofc)

Perhaps im just being special here, but am I missing something here for T2V image training? I've posted the command line below. I have tried previously same settings with runs up to 4.5k steps (195 epochs) - but not had anything better than 40% ish likeness at strength 1. Comparing epoch 30 to 100  in comfyui for example, there's little to no change. Samples wise, I get good/awesome likeness by epoch 30 which persists all the way to 100 with no changes that I can see visually?

Obviously if im using strength of 2.0 however, then everyone looks like the subject sadly, even different genders. I also tripled checked that I was using the proper "comfy" safetensors file as well, in case I was being really special. I'm going to set off another run at fp8, 400 epochs this time which should be about 9,200 steps - so by then it should be absolutely baked I would imagine.

Command Line:

accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 "%mt_path%\ltx2_train_network.py" ^
--mixed_precision bf16 ^
--dataset_manifest _ltx23_lora_dataset_manifest.json ^
--ltx2_checkpoint G:\GenAI\Models\checkpoints\ltx-2.3-22b-dev.safetensors ^
--ltx_version 2.3 ^
--flash_attn ^
--gradient_checkpointing ^
--max_data_loader_n_workers 1 ^
--persistent_data_loader_workers ^
--learning_rate 2e-4 ^
--optimizer_type AdamW8bit ^
--lr_scheduler constant_with_warmup ^
--lr_warmup_steps 10 ^
--network_module networks.lora_ltx2 ^
--network_dim 32 ^
--network_alpha 32 ^
--timestep_sampling shifted_logit_normal ^
--output_dir output ^
--output_name 2e-4-multi-ltx2 ^
--log_config ^
--log_with tensorboard ^
--logging_dir "logs" ^
--log_prefix "2e-4-multi-ltx2" ^
--save_state ^
--save_every_n_epochs 1 ^
--save_last_n_epochs_state 5 ^
--max_train_epochs 400 ^
--sdpa ^
--blocks_to_swap 18 ^
--autoresume

**Comment by AkaneTendo25** [comment]
Concrete settings will vary depending on the dataset, but in your case I'd try lowering the LR to `1e-4` or even `8e-5`, and maybe switching to a broader `lora_target_preset` such as `video_sa_ff` or `video_sa_ca_ff`. The model might learn too 'coarse' with 2e-4 and ignore more subtle features.

**Comment by randomkaos-boop** [comment]
Thanks. I'll give 8e-5 a try without changing anything else, and failing that one, ill try one of the other presets.

But as an FYI, i let it run overnight to epoch 304, which was step 20,907 - loss was about 0.3 based on tensorboard, but there was no difference in comfyui likeness output (compared to epoch 30) - but the video background had started to degrade by that point.

**Comment by dorpxam** [comment]
@AkaneTendo25, I think @randomkaos-boop is right. My own little experience on training with the 2.3 is pretty the same. While the early training with 2.0 I got very clean convergence phase. Here the video-loss seem always straight but seem to learn a little. I say that using various type of optimizer : AdamW, AdamWSchedulerFree, Prodigy ... I've test with some conceptual LoRA but with very simple character LoRA too. Character LoRA is something very simple to train (in principle).

Ofcourse, adding prior divergence or CREPA is not the problem because most of the train was do without. For now, I thought it was a problem related to my own datasets, but what it says @randomkaos-boop it makes me think there might be something more structural.

I'll be frank. I appreciate all your work and dedication, but the accumulation of features and the increasing complexity of the process without systematic and rigorous testing is a challenge. I'm sure advanced users who understand the complex concepts of partial and targeted training will find it helpful, but personally, it's already become too complex for me.

I don't have the skill level or the patience to spend hours and hours blindly manipulating hundreds of parameters for an approximate result. I'll give you a very simple example, with CREPA. The student and the teacher. By default, 16 and 32. Why? How? Do these default values ​​work for everything, or are they just an example?

Anyway. This doesn't call into question your contribution; it's just that I've lost track of things for a while now. I remember my first training session with the "Twin Peaks" dataset, which went like a charm. I can't even train this same dataset anymore and recapture the feeling I had at the beginning.

I know you've focused a lot of your work on audio and animation training, but perhaps we should consider compartmentalizing the process to rediscover what made training the model so simple.

Systematic gradient analysis is one track to explore. It helped me understand that, in reality, the audio practically takes control of the training. That's already a lead. As for image-based training, we now need to figure out why it's become so complicated and why convergence no longer occurs (or at least isn't visible in the loss).

For example, here is an example of what I get in most of the training. 

<img width="2201" height="484" alt="Image" src="https://github.com/user-attachments/assets/f0f6d3d2-3581-4c67-a49e-bf07e7f8af03" />

Something almost constant. And as I told you, regardless of the dataset type, the LR, or how much you optimize it, the loss/video always remains flat and never truly converges.

It might just be me, I admit, I don't presume to question what's being done, but it's a feeling I have. Just as I have the impression that the block swap has an impact on the initial loss. By changing the block swap, under similar conditions, the initial loss can vary considerably. That said, I'm not making a definitive statement. I'll check this point, but perhaps it has an impact on my training.

In any case, I can't thank you enough for all your efforts and your considerable work. I wish I had your work ethic, truly. 👍

**Comment by AkaneTendo25** [comment]
@dorpxam I understand your point, but almost all new features are opt-in. And at its core, all default Musubi-Tuner settings are preserved, except maybe some LTX-specific settings. The training pipeline itself is basically just a modified version of the official trainer code, so I don't think there is some inherent flaws in implementation.

As for extensive testing, I agree that would be the proper way to develop this project, but I just can't physically afford it right now.

**Comment by dorpxam** [comment]
@AkaneTendo25 I understand too, and in any case I say you do a bad job. You know that I'm a huge fan of your work. It's possible that LTX2.3 is more complex to train too.

For the extensive testing, it's the deal but I agree to the complexity. It's a huge task and understand that you can do that yourself. It's just sad to do not have much report here, even when there's no problem.

I think the new estimate feature it's a good point for a better understanding. I probably doing some mistake, for sure. I'm just trying to find the right way to increase my success rate during training sessions. It's just that right now, I'm not managing it 😄 

We are all less and less in a hurry to get good results, but let's not forget that the model is recent and that Lightricks has also made some mistakes: see the 1.1 update of distilled (and latent upscaler).

**Comment by AkaneTendo25** [comment]
Yes, what you are saying is fair. And anyway, I probably won't be adding more features for the sake of features in the near future, and will most likely focus on testing and refining existing stuff (that's why I created this issue as a reminder for myself).
I also have to maintain a few other projects, so my time on this fork is a bit limited, but I'll try to keep it moving.

**Comment by holycowdude** [comment]
@AkaneTendo25 I really appreciate all the updates you've been making

I've trained some LTX2.3 character lora models using your fork and they all seem to work well. (likeness & voice)

The only issue i'm trying to get around is some clothing patterns baking into the character lora by the time the face is learned enough, I think this is because all my dataset videos have the person wearing the same shirt (interview content is rare for this person)
I've described the shirt and pattern in detail in the video text files and i've added an image dataset now with different clothing and that helps a little

if it helps anyone here are my settings which seem to work and train well: (let me know if you think they can be improved!!)

accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 ltx2_train_network.py `
  --mixed_precision bf16 `
  --dataset_config "" `
  --dataset_manifest "" `
  --ltx2_checkpoint "ltx-2.3-22b-dev.safetensors" `
  --gemma_root "gemma-3-12b-it" `
  --ltx_version 2.3 `
  --ltx2_mode av `
  --sdpa `
  --fp8_base `
  --fp8_scaled `
  --blocks_to_swap 22 `
  --gradient_checkpointing `
  --lora_target_preset full `
  --max_data_loader_n_workers 1 `
  --persistent_data_loader_workers `
  --learning_rate 6e-5 `
  --optimizer_type AdamW8bit `
  --lr_scheduler constant_with_warmup `
  --lr_warmup_steps 40 `
  --network_module networks.lora_ltx2 `
  --network_args "train_transformer=True" "train_audio=True" `
  --network_dim 64 `
  --network_alpha 64 `
  --output_dir output `
  --output_name ltx23_V1 `
  --log_config `
  --log_with tensorboard `
  --logging_dir logs `
  --save_state `
  --save_every_n_epochs 5 `
  --max_train_epochs 100 `
  --sample_every_n_epochs 5 `
  --sample_with_offloading `
  --use_precached_sample_prompts `
  --use_precached_sample_latents `
  --sample_prompts _ltx23_lora_sampling.txt `
  --validate_every_n_steps 100 `

**Comment by randomkaos-boop** [comment]
It seems there is a significant difference when I stick to 2e-4 and use `--lora_target_preset full`. Not complete likeness, but much better in ComfyUI now. I'll try a longer run with a lower LR and see how it pans out.

In case anyone comes across this later in life.. its for image training.. T2V.

Edit: Only downside is character bleed across genders, ill try it with DOP as well.

**Comment by holycowdude** [comment]
try without --flash_attn ^ as well and use --sdpa
I heard that can mess with training? but not sure if thats old advice

**Comment by AkaneTendo25** [comment]
There have been reports that some flash_attn setups were broken on RTX 5090. But not _broken_ in the sense that it worsened convergence, it's just LoRA outputted noise. Not your case, I believe.

**Comment by holycowdude** [comment]
@randomkaos-boop 

Could you try without --timestep_sampling shifted_logit_normal ^

I used this command on two training runs and it seemed to mess up the character lora result for me, not sure if it was just a one off so please let me know if it helps (I was training Lora preset full with a video dataset with speech inside the videos)

**Comment by randomkaos-boop** [comment]
Can someone confirm if --network_weights works properly with LoRa's from Civitai for example or merging loras?

I'm currently trying to do base model + civitai lora + character training - and every time I try and use a lora with network_weights it gives me key mismatch errors. I've tried about 10/15 so far.. SFW and NSFW (not that i makes a difference).

I also tried merging 2 lora's with the script in the repo and then using the combined lora with --network_weights and that got the same mismatch errors as well? - yet the lora works fine in ComfyUI ?

However if I use --base_weights instead of network_weights.. I dont get an error at all?

**Comment by AkaneTendo25** [comment]
`--base_weights` and `--network_weights` do different things.

`--network_weights` loads weights into the LoRA network created by the current training command, so the keys, target layers, and rank usually need to match the current `--network_dim`, `--lora_target_preset`, `--network_args`, model version, etc.

`--base_weights` instead builds the LoRA structure from the weight file and merges matching LoRA modules into the base model before training. That is why it can work even when `--network_weights` fails.

So for your "base model + Civitai LoRA + train a new character LoRA" case, `--base_weights` is usually the right option.

If you still want to use `--network_weights`, please paste the exact error or traceback here. This should show the cause of the mismatch.

**Comment by randomkaos-boop** [comment]
Thanks for the reply. 

I figured this out shortly after posting, i should of updated really but was moving on with things. The last time I tried network_weights was some time ago with Flux 1 dev, and im sure it acted differently back then.

But... Using base model + base weights to train a character lora, i found a greater likeness when, after training the character lora side, I then merged the base weights and new character lora together. That improved likeness dramatically with inference - along with keeping the extra functionality from the base weights lora. 

However, I do believe there is an issue with the merge script(s) - merging the base model and a lora together (comfyui format) seems to fail every time. I let Gemini loose on that to sort that, but then came across another error - related to an iterable object or a string - not sure which if im honest, i didnt keep track -  I set Gemini loose on that one again and it seems to of sorted it (probably incorrectly however) and produced a merged model. However, the base model + lora merge doesnt seem to work, or at least doesnt work for inference in ComfyUI.
Originally, when it did "Load Checkpoint" in ComfyUI it failed saying it the dimensions didnt match, so after letting Gemini loose, i got further (properly incorrectly, lets face it) and it failed on the sampler stage of the workflow instead.

When I get chance, ill try it again, but thought i'd mention it in case you want to check it out.
